packages/shared/src/components/PostOptionsMenu.tsx

```diff
@@ -68,34 +68,44 @@ export default function PostOptionsMenu({
     });
 
     if (blockSource) {
-      await unfollowSource({ source: reportedPost?.source });
+      await onUnfollowSource({ source: reportedPost?.source });
     }
 
     showMessageAndRemovePost('🚨 Thanks for reporting!', reportPostIndex);
   };
 
   const onBlockSource = async (): Promise<void> => {
-    await unfollowSource({ source: post?.source });
+    await onUnfollowSource({ source: post?.source });
     showMessageAndRemovePost(`🚫 ${post?.source?.name} blocked`, postIndex);
   };
 
   const onBlockTag = async (tag: string): Promise<void> => {
-    await blockTag({ tags: [tag] });
+    await onBlockTags({ tags: [tag] });
     showMessageAndRemovePost(`⛔️ #${tag} blocked`, postIndex);
   };
 
   const onHidePost = async (): Promise<void> => {
     const promise = hidePost(post.id);
     await Promise.all([promise, onRemovePost?.(postIndex)]);
-    onMessage(
-      '🙈 This article won’t show up on your feed anymore',
-      postIndex,
-      0,
-    );
+    if (!postIndex) {
+      onMessage(
+        '🙈 This article won’t show up on your feed anymore',
+        postIndex,
+        0,
+      );
+    }
   };
 
   const onSharePost = async () => {
-    await navigator.clipboard.writeText(post.permalink);
+    trackEvent({
```

Comment:
I created a helper function for post events that you need to use to be consistent with the other post events.
I don't remember the name but look for it

Reply:
@idoshamun Do you mean this one?

```js
trackEvent(
      postAnalyticsEvent('share post', post, {
        extra: { media, origin: 'share bar' },
      }),
    );
```

Reply:
Exactly! Another thing I noticed now when the native share is supported it should be used instead of the clipboard.
I think I have some helpers around somewhere in the article page components.